kufar/books/main.go

```diff
@@ -107,34 +113,16 @@ func main() {
 		r.Get("/", ReadBooks(*db))
 
 		r.Route("/{id}", func(r chi.Router) {
-			r.Get("/", ReadBook(*db))
+			c := NewCache()
```

Comment:
Please move your init upper, before all working with router. Try to divide your main function into some logical parts, e.g. config reading, initializations, route building, server start, server stop and so on